errors.go

```diff
@@ -11,41 +11,46 @@ import (
 type (
 	// Errors may be generated during deserialization, binding,
 	// or validation. It implements the built-in error interface.
-	Errors []Error
+	Errors []FieldError
 
 	// Error is a powerful implementation of the built-in error
 	// interface that allows for error classification, custom error
 	// messages associated with specific fields, or with no
 	// associations at all.
-	Error struct {
-		// An error supports zero or more field names, because an
-		// error can morph three ways: (1) it can indicate something
-		// wrong with the request as a whole, (2) it can point to a
-		// specific problem with a particular input field, or (3) it
-		// can span multiple related input fields.
-		FieldNames []string `json:"fieldNames,omitempty"`
+	fieldError struct {
+		// An error supports a field names because an
+		// error can morph two ways: (1) it can indicate something
+		// wrong with the request as a whole, or (2) it can point to a
+		// specific problem with a particular input field.
+		fieldName string `json:"fieldNames,omitempty"`
```

Comment:
When I created the binding library, it was the only one - and I think still is, from what I've seen - the only one that allows us to map an error to zero or more field names, which is a requirement that our company had when I was still working there. Is there a way we can preserve this feature? It definitely has its advantages.